File: 2025/04 April/22 Unique Number I/main.py
# Link: https://www.geeksforgeeks.org/problems/find-unique-number/1


class Solution:
    def findUnique(self, arr):
        x = 0
        for e in arr:
            x = x ^ e
        return x


if __name__ == "__main__":
    obj = Solution()
    arr = [1, 2, 1, 5, 5]
    print(obj.findUnique(arr))

    arr = [2, 30, 2, 15, 20, 30, 15]
    print(obj.findUnique(arr))

File: 2025/04 April/22 Unique Number I/main.cpp
// Link: https://www.geeksforgeeks.org/problems/find-unique-number/1

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int findUnique(vector<int> &arr)
    {

        int res = 0;
        for (int i = 0; i < arr.size(); i++)
        {
            res ^= arr[i];
        }
        return res;
    }
};

int main()
{
    Solution obj;
    vector<int> arr = {1, 2, 1, 5, 5};
    cout << "The unique number is: " << obj.findUnique(arr) << endl;

    arr = {2, 30, 2, 15, 20, 30, 15};
    cout << "The unique number is: " << obj.findUnique(arr) << endl;

    return 0;
}